### PR TITLE
DS-883 | @ericbakenhus | Use Luxon for all the things

### PR DESCRIPTION
**Pull request by ericbakenhus**
# READY FOR REVIEW

# Summary
- Swap all date utility function to use Luxon

# Review By (Date)
- 🤷 

# Criticality
- Medium

# Review Tasks

## Setup tasks and/or behavior to test

1. Look at code
2. Check out this branch
3. Make sure trip dates still look correct, for example:
    - Go to one of the trips (e.g. https://deploy-preview-951--stanford-alumni.netlify.app/travel-study/destinations/finland-2022/), get to the GiveGab form, and check the variables set on Window to make sure the dates look alright

# Associated Issues and/or People
- DS-883

### CODE REVIEW

File: src/utilities/transformDate.js [hunk 1]
```diff
@@ -1,26 +1,21 @@
+import { DateTime } from 'luxon';
+import { luxonDate } from './dates';
+
 export const formatFmDate = (tripDate) => {
-  const date = new Date(tripDate).toLocaleDateString('en-US');
+  const date = luxonDate(tripDate).toLocaleString();
   return date;
 };
```
Comment (ericbakenhus) [inline src/utilities/transformDate.js:7]: Just swaps over to `luxonDate`. `luxonDate` sets the locale to `en-US` so `toLocaleString` uses that by default.

File: src/utilities/transformDate.js [hunk 2]
```diff
@@ -1,26 +1,21 @@
...
 export const formatEmailDate = (tripDate) => {
-  const dateFormat = {
-    month: 'long',
-    day: 'numeric',
-    year: 'numeric',
-  };
-  const date = new Date(tripDate).toLocaleDateString('en-US', dateFormat);
+  const date = luxonDate(tripDate).toLocaleString(DateTime.DATE_FULL);
   return date;
 };
```
Comment (ericbakenhus) [inline src/utilities/transformDate.js:12]: `DateTime.DATE_FULL` results in the same format (`MMMM d, yyyy`) as the manually defined one.

File: src/utilities/transformDate.js [hunk 3]
```diff
@@ -1,26 +1,21 @@
...
+  if (!date?.isValid) {
     return undefined;
   }
-  const month = date.getUTCMonth() + 1;
-  const day = date.getUTCDate();
-  const year = date.getUTCFullYear();
-  const newDate = `${month}/${day}/${year}`;
+  const newDate = date.toLocaleString();
   return newDate;
 };
```
Comment (ericbakenhus) [inline src/utilities/transformDate.js:21]: `M/d/yyyy` is the same as the default en-US locale format.

**Comment by ericbakenhus** [comment]
@sherakama There's also some date poking in [gg-trip-form.js](https://github.com/SU-SWS/saa_alumni/blob/dev/static/scripts/travel-study/gg-trip-form.js). I don't know what's going on with that script though, so I'm not going to mess with it.

**Comment by sherakama** [comment]
> There's also some date poking in [gg-trip-form.js](https://github.com/SU-SWS/saa_alumni/blob/dev/static/scripts/travel-study/gg-trip-form.js?rgh-link-date=2024-09-10T23%3A14%3A16Z). I don't know what's going on with that script though, so I'm not going to mess with it.

That script is the embed widget for the intranet (corral). That site is primarily staff facing so we shouldn't have the same browser issues there.

**Comment by sherakama** [comment]
Screenshot 2024-09-11 at 09 39 02

**Pull request merged**